py3status/modules/updates.py

```diff
@@ -0,0 +1,393 @@
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x[:4] == "Inst"])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Cargo(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[2:])
+
+
+class Dnf(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Npm(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Pakku(Update):
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError as ce:
+            return ce.output
+
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x == "aur"])
+
+
+class Pip(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[2:])
+
+
+class Pkcon(Update):
+    def get_output(self):
+        try:
+            output = self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError:
+            return ""
+        return output.partition("Results:\n")[-1]
+
+
+class Trizen(Update):
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError as ce:
+            return ce.output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    managers = []
+    thresholds = [
+        (0, "darkgray"),
+        (10, "degraded"),
+        (20, "orange"),
+        (30, "bad"),
+    ]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                },
+                {
+                    "placeholder": "total",
+                    "new": "update",
+                    "format_strings": ["format"],
+                },
+                {
+                    "placeholder": "updates",
+                    "new": "update",
+                    "format_strings": ["format"],
+                },
+            ]
+        }
+
+    def post_config_hook(self):
+        with open("/etc/os-release") as f:
+            multiple = "archlinux" in f.read()
```

Comment:
we should check the id field here to better differentiate between distros and avoid unneeded runs of the which command.

Reply:
I don't think that would happen, but I'd like to be proved wrong first.

Reply:
well the id is the short distro name in lowercase so 'arch','debian','fedora','ubuntu'...

It is not relay important but you started with detecting arch as distro so if we do something like this we should do it right. As one could also remove this check and wait for the check_command farther down.

Reply:
>As one could also remove this check and wait for the check_command farther down.

I had this to detect Arch to determine if we should stop or keep going in the `managers` list. With Arch Linux, I try all supported AUR helpers without stopping. With other distributions, I try until I find one and stop there. 

Also, it's "faster" for all if I keep Arch part separated. This is now more separated to the point where I wonder if we will run into (minor) issue such as `yum`, `dnf` could be used on more than one distribution, but I'm only checking `fedora` and not others. Also, we need to check/compare several `/etc/os-releases` too.  Also, `ID_LIKE` is not always guaranteed in `/etc/os-release`.